axon/src/handler.rs

```diff
@@ -90,7 +92,7 @@ impl IbcChannel {
         {
             return false;
         }
-        return true;
+        true
```

Comment:
Nit: just use `!a` instead of `if a {return false} true`.

Personally I'd write this as `(self.num, self.port_id, self.order, self.sequence) == (other.num, other.port_id, other.order, other.sequence)`, taking advantage of [tuple equality](https://doc.rust-lang.org/stable/std/cmp/trait.Eq.html#impl-Eq-for-(T,)).